### DISCUSSION TITLE
Help to serve .pub file with content-type "text/plain" for AMP site

### DISCUSSION THREAD

**Discussion by fernandohlee** [vercel/vercel#4891] [Help] [answered]
Hi, I have an AMP site and I'm trying to use the update-cache request to invalidate some pages cache.

	https://developers.google.com/amp/cache/update-cache#call-the-update-cache

I'm already serving the file:

	you must serve the public RSA key at a fixed location on the AMP document's domain  For example:
	https://example.com/.well-known/amphtml/apikey.pub

However, I'm stuck in one of the guidelines, which is:

	You must publish the key in PEM format and serve the key with the content-type "text/plain".

I have no idea how to do that in VERCEL or maybe with some Nextjs configuration, any help is appreciated.

**Comment by styfle** [comment, marked as answer]
You can add a `vercel.json` file with the `headers` property to change the content type.

```json
{
  "headers": [
    {
      "source": "/.well-known/amphtml/apikey.pub",
      "headers": [
        {
          "key": "Content-Type",
          "value": "text/plain"
        }
      ]
    }
  ]
}
```

https://vercel.com/docs/configuration#project/headers

I think we can change the platform to always serve `.pub` as `text/plain`, but until then this config will work. Thanks!
